docs/source/query/overview.rst

```diff
@@ -0,0 +1,216 @@
+Sentry can and should be used to send the query to Snuba.
+
+The api is in `this module <https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L667>`_.
+It takes care of caching, retries and allows bulk queries.
+
+The method returns a dictionary that contains the data in response and
+additional metadata::
+
+    {
+        "data": [
+            {
+                "title": "very bad",
+                "uniq_events": 2
+            }
+        ],
+        "meta": [
+            {
+                "name": "title",
+                "type": "String"
+            },
+            {
+                "name": "uniq_events",
+                "type": "UInt64"
+            }
+        ],
+        "timing": {
+            ... details ...
+        }
+    }
+
+The ``data`` section is a list with one dictionary per row. The ``meta``
+section contains the list of the columns included in the response with
+their data type as inferred by Clickhouse.
+
+More details about the structure of the timing section below.
+
+Sending a test query through the web UI
+=======================================
+
+Snuba has a minimal web UI you can use to send queries. You can run Snuba
+locally and the web UI will be accessible at ``http://localhost:1218/[DATASET NAME]/snql``.
+
+.. image:: /_static/query/snubaUI.png
+
+The SnQL query should be provided (sorry, on one line only) in the `query`
+attribute and the structure of the response is the same discussed in the
+section above.
+
+Sending a query via curl
+========================
+
+The web ui just sends the payload as a POST. So the same result can be
+achieved with curl or any other HTTP client.
+
+Request and response formats
+============================
+
+The request format is the same visible in the screenshot:
+
+* ``query`` contains the SnQL query as a string
+* ``dataset`` is the dataset name (if not already specified in the url
+* ``debug`` makes Snuba provide exhaustive statistics in the response
+            including the Clickhouse query.
+* ``consistent`` forces the Clickhouse query to be executed in single
+                 threaded mode and, in case the Clickhouse table is
+                 replicated, it will force Snuba to always hit the same
+                 node. Which can guarantee sequential consistency as
+                 that is the node where the consumer write by default.
+                 This is achieved with the `load balancing <https://clickhouse.tech/docs/en/operations/settings/settings/#load_balancing-in_order>`_
+                 Clickhouse property which is set as ``in_order``.
+* ``turbo`` sets a sampling rate to the query defined in the ``TURBO_SAMPLE_RATE``
+            Snuba setting. It also prevents Snuba to apply the ``FINAL``
+            mode to the Clickhouse query in case it was needed to guarantee
+            correct results after replacements.
+
+
+The response format is the same discussed above. The complete version
+looks like this (in debug mode) ::
```

Comment:
I would either have a section on errors or at minimum mention the "error" section of the response here.

Reply:
I added details around errors